### ISSUE TITLE
Force generated parameters to be defined using $ref pointing at the proto definition

### ISSUE THREAD

**Issue by njerschow** [grpc-ecosystem/grpc-gateway#2815]
### Current behavior: 
All generated paths result in the request parameter definitions looking like this:
```json
"parameters": [
          {
            "name": "pool_id",
            "in": "path",
            "required": true,
            "type": "string",
            "format": "uint64"
          }
        ],
```

It would be awesome if there were a flag that could force my parameters to be defined using a ref instead:
```json
"parameters": [{
            "name": "body",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/com.example.myRequestDefinitionWithPoolId"
            }
          }]
```
          
### Why?
I have a relatively niche use case (I think) where I need to be able to send arbitrary RPC messages without knowing what type they will be ahead of time. aka there are 100s of requests, if I could just define them using JsonSchema and then map them properly to the grpc-gateway requests my life would be a lot easier.

Why can't I just use the parameters? Because the requests are defined with a UI conforming to the RPC message type, but I'd like to use gateway requests so as not to need to build a full RPC server.

If anyone could point me to a place in the code where I can potentially make this change, that would be greatly appreciated & I will try to submit a PR (i am a go newb).

alternatively, if the feature already exists, then feel free to close this, though I'm pretty sure it does not.

**Comment by johanbrandhorst** [comment]
Hi Nikita, thanks for your issue! Parameters will use a `$ref` whenever possible, however in your use case it might not be happening because part of the message is inferred from the path and part from the body. In such cases, the gateway will remove the parameters that are inferred from the path from the body, and inline the message. This ensures that the generated schema matches what the server behavior is.

In other words, if you use _no_ path parameters, and use `body: *` in your annotations (and you're using POST requests), you should get the `$ref` version rendered.

Having said all of that, your use case is one where protobuf does not work very well. Things a always easier when the messages are known ahead of time, so maybe the grpc-gateway isn't the best solution for your problem.

**Comment by njerschow** [comment]
Thanks for the answer johan, I will look to implement this in other ways.
